harmonica/_io/oasis_montaj_grd.py

```diff
@@ -0,0 +1,422 @@
+        # Decompress the block
+        grid_sub = zlib.decompress(
+            grid_compressed[start_offset:end_offset],
+            bufsize=zlib.DEF_BUF_SIZE,
+        )
+        # Add it to the running grid
+        grid += grid_sub
+    return grid
+
+
+def _build_coordinates(west, south, shape, spacing):
+    """
+    Create the coordinates for the grid
+
+    Generates 1d arrays for the easting and northing coordinates of the grid.
+    Assumes unrotated grids.
+
+    Parameters
+    ----------
+    west : float
+        Westernmost coordinate of the grid.
+    south : float
+        Southernmost coordinate of the grid.
+    shape : tuple
+        Tuple of ints containing the number of elements along each direction in
+        the following order: ``n_northing``, ``n_easting``
+    spacing : tuple
+        Tuple of floats containing the distance between adjacent grid elements
+        along each direction in the following order: ``s_northing``,
+        ``s_easting``.
+
+    Returns
+    -------
+    easting : 1d-array
+        Array containing the values of the easting coordinates of the grid.
+    northing : 1d-array
+        Array containing the values of the northing coordinates of the grid.
+    """
+    easting = np.linspace(west, west + spacing[1] * (shape[1] - 1), shape[1])
+    northing = np.linspace(south, south + spacing[0] * (shape[0] - 1), shape[0])
+    return easting, northing
+
+
+def _build_rotated_coordinates(west, south, shape, spacing, rotation_deg):
+    """
+    Create the coordinates for a rotated grid
+
+    Generates 2d arrays for the easting and northing coordinates of the grid.
+    Assumes rotated grids.
+
+    Parameters
+    ----------
+    west : float
+        Westernmost coordinate of the grid.
+    south : float
+        Southernmost coordinate of the grid.
+    shape : tuple
+        Tuple of ints containing the number of elements along each unrotated
+        direction in the following order: ``n_y``, ``n_x``
+    spacing : tuple
+        Tuple of floats containing the distance between adjacent grid elements
+        along each unrotated direction in the following order: ``spacing_y``,
+        ``spacing_x``.
+
+    Returns
+    -------
+    easting : 2d-array
+        Array containing the values of the easting coordinates of the grid.
+    northing : 2d-array
+        Array containing the values of the northing coordinates of the grid.
+
+    Notes
+    -----
+    The ``x`` and ``y`` coordinates are the abscissa and the ordinate of the
+    unrotated grid before the translation, respectively.
+    """
+    # Define the grid coordinates before the rotation
+    x = np.linspace(0, spacing[1] * (shape[1] - 1), shape[1])
+    y = np.linspace(0, spacing[0] * (shape[0] - 1), shape[0])
+    # Compute a meshgrid
+    x, y = np.meshgrid(x, y)
+    # Rotate and shift to get easting and northing
+    cos = np.cos(np.radians(rotation_deg))
+    sin = np.sqrt(1 - cos**2)
```

Comment:
I assume that should be sin = -np.sin(np.radians(rotation_deg))
sin = np.sqrt(1 - cos**2) give same value for positve and negtive rotation angle

Reply:
Nice catch @LL-Geo! Now it's fixed. Have you visually check if the rotated grid is read as expected?

Running this should generate the same plot as the one you get in Oasis:
```python
import harmonica as hm
import matplotlib.pyplot as plt

grid = hm.load_oasis_montaj_grid("harmonica/tests/data/om_rotate.grd")
grid.plot(x="easting", y="northing")
plt.show()
```